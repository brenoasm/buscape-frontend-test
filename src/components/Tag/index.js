import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from '../../theme';

const propTypes = {
  text: PropTypes.string.isRequired
};

const defaultProps = {
  text: ''
};

const StyledTag = styled.span`
  font-size: 10px;
  text-align: center;
  color: ${colors.white};
  position: relative;
  background: ${colors.blue};
  border-radius: 2px;
  padding: 5px 10px;

  &:after {
    content: "";
    position: absolute;
    left: 0;
    bottom: 0;
  }

  &:before {
    content: "";
    position: absolute;
    right: -10px;
    bottom: 0;
    border-left: 10px solid ${colors.blue};
    border-top: 10px solid transparent;
    border-bottom: 13px solid transparent;
  }
`;

const Tag = ({
  text
}) => (
  <StyledTag>
    {text}
  </StyledTag>
);

Tag.propTypes = propTypes;
Tag.defaultProps = defaultProps;

export default Tag;
