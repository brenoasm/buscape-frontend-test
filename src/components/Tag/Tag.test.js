import React from "react";
import { shallow } from "enzyme";

import Tag from "./";

describe("Tag", () => {
  it("matches the snapshot", () => {
    const wrapper = shallow(<Tag text="Teste" />);

    expect(wrapper).toMatchSnapshot();
  });
});
