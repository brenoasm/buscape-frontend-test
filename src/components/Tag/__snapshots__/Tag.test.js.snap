// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tag
    text="Teste"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Teste",
    },
    "ref": null,
    "rendered": "Teste",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  font-size: 10px;
  text-align: center;
  color: ",
          "#FFF",
          ";
  position: relative;
  background: ",
          "#1990DF",
          ";
  border-radius: 2px;
  padding: 5px 10px;

  &:after {
    content: \\"\\";
    position: absolute;
    left: 0;
    bottom: 0;
  }

  &:before {
    content: \\"\\";
    position: absolute;
    right: -10px;
    bottom: 0;
    border-left: 10px solid ",
          "#1990DF",
          ";
    border-top: 10px solid transparent;
    border-bottom: 13px solid transparent;
  }
",
        ],
      },
      "displayName": "styled.span",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Teste",
      },
      "ref": null,
      "rendered": "Teste",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  font-size: 10px;
  text-align: center;
  color: ",
            "#FFF",
            ";
  position: relative;
  background: ",
            "#1990DF",
            ";
  border-radius: 2px;
  padding: 5px 10px;

  &:after {
    content: \\"\\";
    position: absolute;
    left: 0;
    bottom: 0;
  }

  &:before {
    content: \\"\\";
    position: absolute;
    right: -10px;
    bottom: 0;
    border-left: 10px solid ",
            "#1990DF",
            ";
    border-top: 10px solid transparent;
    border-bottom: 13px solid transparent;
  }
",
          ],
        },
        "displayName": "styled.span",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
